### PR TITLE
Make sure final member status is Removed, #31030

### PR DESCRIPTION
**Pull request by patriknw**
* race condition between the close() setting volatile _closed flag
  and the eventBusListener already having a MemberExited message in flight

References #31030

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterReadView.scala [hunk 1]
```diff
@@ -187,9 +190,15 @@ private[akka] class ClusterReadView(cluster: Cluster) extends Closeable {
    */
   def close(): Unit = {
     _closed = true
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterReadView.scala:192]: I could reproduce the test failure by adding a strategic sleep in the eventBusListener actor

**Review by johanandren — APPROVED** [review]

**Review by jrudolph — COMMENTED** [review]
Seems a bit arbitrary to me. Why should have closing this view have an effect on the final state of the member status?

And why fix that particular bit? The class gives no guarantees and there are still lots of potential remaining inconsistencies because updates are not atomic...

**Comment by patriknw** [comment]
> Why should have closing this view have an effect on the final state of the member status?

Because there is only one and it has the same lifecycle as the Cluster extension instance. I agree that it's not idomatic state management at all, since it has state outside the actor for synchronous (cached) access.

I can rewrite it a bit (without changing too much) by only update the state from inside the actor.

**Comment by patriknw** [comment]
state mgmt updated in https://github.com/akka/akka/pull/31144/commits/ee41aa3cd0b2c78e1a809e65320dc752a7857aa4

**Review by jrudolph — APPROVED** [review]
LGTM, it's a good step forward. The calling code still has no guarantee that the results are consistent across calls but returning something like a "snapshot" can now be added more easily if needed.

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
Created https://github.com/akka/akka/issues/31150 for the test failure, which I think is unrelated.

**Pull request merged**